crates/sui-node/src/lib.rs

```diff
@@ -641,19 +633,22 @@ impl SuiNode {
     /// This function waits for a signal from the checkpoint executor to indicate that on-chain
     /// epoch has changed. Upon receiving such signal, we reconfigure the entire system.
     pub async fn monitor_reconfiguration(self: Arc<Self>) -> Result<()> {
+        let mut checkpoint_executor = CheckpointExecutor::new(
+            self.state_sync.subscribe_to_synced_checkpoints(),
+            self.checkpoint_store.clone(),
+            self.state.clone(),
+            self.config.checkpoint_executor_config.clone(),
+            &self.registry_service.default_registry(),
+        )?;
+
```

Comment:
@lxfind notice here that we move checkpoint executor out of the loop so that we do not re-instantiate every time. This should be safe due to the following: 

1. The only relevant state that executor manages is the receive channel from state sync, which is fine to live across epochs (state sync is epoch agnostic)
2. The authority state doesn't get changed between epochs - only its epoch store does. So maintaining this reference should be fine

What we get from this is the same startup speed as if we were to not have executor run per epoch, while also removing the race conditions around epoch store (since suinode still restarts executor at the beginning of the new epoch)